Test for POST /users/login/switch fail on 4.0

```
Aug 07, 2017 1:03:29 PM org.glassfish.jersey.logging.LoggingInterceptor log
SEVERE: 19 * Sending client request on thread ForkJoinPool.commonPool-worker-29
19 > POST http://localhost:8065/api/v4/users/login/switch
19 > Accept: application/json
19 > Authorization: Bearer 4tjmxaxtjprr8cgiqfe7tu6djo
19 > Content-Type: application/json
{"current_service":"email","new_service":"gitlab","email":"success+zf8yr68txmz6nxmeuu9ravvxqo@inbucket.local","current_password":"Password1","new_password":null,"mfa_code":null,"ldap_id":null}

Aug 07, 2017 1:03:29 PM org.glassfish.jersey.logging.LoggingInterceptor log
SEVERE: 19 * Client response received on thread ForkJoinPool.commonPool-worker-29
19 < 401
19 < Content-Length: 188
19 < Content-Type: application/json
19 < Date: Mon, 07 Aug 2017 13:03:29 GMT
19 < X-Request-Id: 8nb76gsshffmzqytaqucruzkxr
19 < X-Version-Id: 4.0.0.4.0.2.0cf768b2704fe0f223937b5a852f45fe.false
{"id":"api.user.check_user_password.invalid.app_error","message":"Login failed because of invalid password","detailed_error":"","request_id":"8nb76gsshffmzqytaqucruzkxr","status_code":401}

testUsers_SwitchLoginMethod(net.bis5.mattermost.client4.MattermostApiTest)  Time elapsed: 0.731 sec  <<< ERROR!
java.util.concurrent.ExecutionException: java.lang.AssertionError: 
Expected: is <501>
     but: was <401>
	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1895)
	at net.bis5.mattermost.client4.MattermostApiTest.testUsers_SwitchLoginMethod(MattermostApiTest.java:1134)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: java.lang.AssertionError: 
Expected: is <501>
     but: was <401>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at net.bis5.mattermost.client4.MattermostApiTest.checkStatus(MattermostApiTest.java:124)
	at net.bis5.mattermost.client4.MattermostApiTest.lambda$testUsers_SwitchLoginMethod$17(MattermostApiTest.java:1133)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:602)
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:577)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1595)
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1582)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

--- a/mattermost-models/src/main/java/net/bis5/mattermost/model/SwitchRequest.java
+++ b/mattermost-models/src/main/java/net/bis5/mattermost/model/SwitchRequest.java
@@ -35,7 +35,10 @@ public class SwitchRequest {
 	private AuthService newService;
 	@JsonProperty("email")
 	private String email;
+	/** @deprecated Mattermost 4.0 or lator: use {@link #password} instead. */
 	@JsonProperty("current_password")
+	private String currentPassword;
+	@JsonProperty("password")
 	private String password;
 	@JsonProperty("new_password")
 	private String newPassword;

--- a/mattermost4j/src/test/java/net/bis5/mattermost/client4/MattermostApiTest.java
+++ b/mattermost4j/src/test/java/net/bis5/mattermost/client4/MattermostApiTest.java
@@ -1130,13 +1130,16 @@ public class MattermostApiTest {
 		assertThat(result, is(true));
 	}
 
+	@SuppressWarnings("deprecation")
 	@Test
 	public void testUsers_SwitchLoginMethod() throws InterruptedException, ExecutionException {
 		SwitchRequest request = new SwitchRequest();
 		request.setCurrentService(AuthService.Email);
 		request.setNewService(AuthService.GitLab);
 		request.setEmail(th.basicUser().getEmail());
-		request.setPassword(th.basicUser().getPassword());
+		request.setCurrentPassword(th.basicUser().getPassword());
+		request.setPassword(th.basicUser().getPassword()); // for 4.0+
+		th.loginBasic();
 
 		client.switchAccountType(request)
 				.thenApply(r -> checkStatus(r, Status.NOT_IMPLEMENTED))

